must-gather/collection-scripts/gather_ceph_resources

```diff
@@ -245,15 +245,16 @@ for ns in $namespaces; do
 
         # Collecting snapshot info for ceph rbd volumes
         printf "collecting snapshot info for ceph rbd volumes \n" | tee -a  "${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
-        COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/rbd_vol_and_snap_info
+        COMMAND_OUTPUT_FILE_VOL=${COMMAND_OUTPUT_DIR}/rbd_vol_info
+        COMMAND_OUTPUT_FILE_SNAP=${COMMAND_OUTPUT_DIR}/rbd_snap_info
         # Inspecting ceph block pools for ceph rbd
         blockpools=$(timeout 60 oc get cephblockpools.ceph.rook.io -n openshift-storage -o jsonpath="{range .items[*]}{@.metadata.name}{'\n'}{end}")
         for bp in $blockpools; do
             images=$(timeout 60 oc -n openshift-storage exec "${HOSTNAME}"-helper -- bash -c "rbd ls -p $bp")
             for image in $images; do
-                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd info $image --pool $bp" >> "${COMMAND_OUTPUT_FILE}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-"${image}"-debug.log 2>&1 &
+                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd info $image --pool $bp" >> "${COMMAND_OUTPUT_FILE_VOL}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-"${image}"-debug.log 2>&1 &
```

Comment:
I don't think this fixes the issue.  Having a single `rbd_vol_and_snap_info` file is actually useful -- the issue is how that file is generated.

Let's ignore snapshots for a bit and concentrate on `rbd info` alone.  You are launching potentially hundreds or thousands of `rbd info` processes concurrently while _appending_ their output to the same file.  These processes are almost certainly going to race with each other, resulting in interleaved output.  For example, in this sample, between two complete outputs for `csi-vol-4d582d70-ac0b-11ec-9225-0a580a80021a` and `csi-vol-4a3e42a3-ac0b-11ec-9225-0a580a80021a` there is a partial output for `csi-vol-4af56d5d-ac0b-11ec-9225-0a580a80021a`:

```
rbd image 'csi-vol-4d582d70-ac0b-11ec-9225-0a580a80021a':
	size 147 GiB in 37632 objects
	order 22 (4 MiB objects)
	snapshot_count: 3
	id: 7a906c55977b
	block_name_prefix: rbd_data.7a906c55977b
	format: 2
	features: layering, exclusive-lock, object-map, fast-diff
	op_features: 
	flags: 
	create_timestamp: Fri Mar 25 07:15:08 2022
	access_timestamp: Mon Mar 28 17:02:01 2022
	modify_timestamp: Fri Mar 25 07:15:08 2022
	mirroring state: enabled
	mirroring mode: snapshot
	mirroring global id: 0c1d44b8-bf48-427f-bf75-4e1224e16d93
	mirroring primary: true
rbd image 'csi-vol-4af56d5d-ac0b-11ec-9225-0a580a80021a':
	size 86 GiB in 22016 objects
rbd image 'csi-vol-4a3e42a3-ac0b-11ec-9225-0a580a80021a':
	size 82 GiB in 20992 objects
	order 22 (4 MiB objects)
	snapshot_count: 3
	id: 7a9067985503
	block_name_prefix: rbd_data.7a9067985503
	format: 2
	features: layering, exclusive-lock, object-map, fast-diff
	op_features: 
	flags: 
	create_timestamp: Fri Mar 25 07:15:03 2022
	access_timestamp: Mon Mar 28 02:47:28 2022
	modify_timestamp: Fri Mar 25 07:15:03 2022
	mirroring state: enabled
	mirroring mode: snapshot
	mirroring global id: bf66b997-ed6c-4ee3-af78-e60087d34b1a
	mirroring primary: true
```

The proper fix would be to direct the output of each command to a separate file and merge them into a single `rbd_vol_and_snap_info` after all `rbd info` and `rbd snap ls --all` processes complete.

Reply:
I am not sure storing each output in a separate file will be preferable or not. But will check on this and update the PR.

Reply:
merging can also be done. at the end by selection only rbd files and using notation `>>`. But a better way would be to use wait and timeout the backend process.